### PR TITLE
Allow applications to require a minimum key mask for authorization.

### PR DESCRIPTION
**Pull request by Acidity**

### CODE REVIEW

File: brave/core/util/eve.py [hunk 1]
```diff
@@ -389,6 +389,9 @@ def __repr__(self):
     def has_access(self, mask):
         if isinstance(mask, EVEKeyMask):
             mask = mask.mask
+        # If the mask we're checking for is greater than this object's mask it has to be False
+        if mask > self.mask:
```
Comment (jboning) [inline brave/core/util/eve.py]: this is not how one compares bitvectors :P

Comment (jboning) [inline brave/core/util/eve.py]: Though, it might be worth adding comparison methods to the EVEKeyMask class so something like this would be correct

Comment (Acidity) [inline brave/core/util/eve.py]: I think you may have missed the part below this... This is just to fix a bug where if we had a key with mask A trying to see if it had permissions for mask B, A & B always returns true if B > A even though A does not give permissions for B.

Comment (jboning) [inline brave/core/util/eve.py]: Ahh. Well, it's still not going to be right, for example mask=0b101, self.mask=0b110.

Comment (Acidity) [inline brave/core/util/eve.py]: yeah, math is hard...

Comment (jboning) [inline brave/core/util/eve.py]: theeere we go :) go ahead and clean up this check again while we're here

File: brave/core/controller.py [hunk 2]
```diff
@@ -56,14 +56,22 @@ def get(self, ar):
...
                 dict(success=False, message=_("This application requires that you have a character connected to your"
                                               " account. Please <a href=\"/key/\">add an API key</a> to your account."),
                      ar=ar))
-            return 'brave.core.template.authorize', dict(success=True, ar=ar, characters=characters, default=default)
+            chars = []
+            for c in characters:
+                if c.credential_for(ar.application.mask.required):
+                    chars.append(c)
+            if chars:
+                default = u.primary or characters[0]
```
Comment (jboning) [inline brave/core/controller.py]: Does this work out right if u.primary is not in characters?

**Pull request merged**